src/Contexts/QuizContext.tsx

```diff
@@ -0,0 +1,36 @@
+import React, { createContext, useContext, useReducer, useEffect } from "react";
+import { Prop, QuizState, QuizContextType } from "./Context";
+import { quizReducer } from "../Reducers/quizReducer";
+import axios from "axios";
+import { useAuth } from "./index";
+
+const QuizContext = createContext({} as QuizContextType)
+
+export function QuizContextProvider({ children } :Prop){
+
+    const initialQuizState: QuizState = { quiz: [], leaderBoard: [] };
+    const [quizState, quizDispatch] = useReducer(quizReducer, initialQuizState);
+    const { authState } = useAuth();
+
+    useEffect(()=>{
+        (async function(){
+            try {
+                const [quiz, leaderboard] = await Promise.all([axios.get("https://quiz-app-backend.ashishgupta08.repl.co/quiz", { headers: { Authorization: authState.token } }), axios.get("https://quiz-app-backend.ashishgupta08.repl.co/leaderBoard", { headers: { Authorization: authState.token } })])
```

Comment:
creating one function which returns baseURL will be helpful and you don't have to type it everywhere